s3tests_boto3/functional/test_iam.py

```diff
@@ -329,3 +329,112 @@ def test_get_deleted_user_policy():
                            UserName=get_tenant_user_id())
 
     client.get_user_policy(PolicyName='AllowAccessPolicy2', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='delete')
+@attr(operation='Verify Delete User Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_delete_user_policy():
+    client = get_tenant_iam_client()
+
+    policy_document_allow = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Allow",
+             "Action": "*",
+             "Resource": "*"}}
+    )
+
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+
+    client.delete_user_policy(PolicyName='AllowAccessPolicy', UserName=get_tenant_user_id())
+
+    e = assert_raises(ClientError, client.get_user_policy, PolicyName='AllAccessPolicy',
+                      UserName=get_tenant_user_id())
+    status = _get_status(e.response)
+    eq(status, 404)
```

Comment:
is this a success case? in that case will the return code be 404?

Reply:
This is not a success case. we are deleting the user policy and then trying to get the user policy.
In that case it will return 404.

Reply:
ok.